### DISCUSSION TITLE
Trying to draw an image turns into an execption

### DISCUSSION THREAD

**Discussion by tails1154** [CosmosOS/Cosmos#3067] [Q&A] [answered]
So when I try to use this code:
```

                try
                {
                    screen.DrawFilledRectangle(Color.Green, 564, 713, 100, 50);
                    screen.DrawImage(drivers.smileyface, 564, 713, 1, 1);
                    //screen.DrawString("Settings", PCScreenFont.Default, Color.Black, 600, 736);
                }
                catch (Exception ex)
                {
                    startupLoader("Exception Occoured" + ex.Message);
                }

```
it gives the exception of Arg_OutOfRangeException. Any ideas?

**Comment by Szymekk44** [comment]
Are you sure your bitmap is correct?
you should load it like this: Bitmap bmp = new Bitmap(bytes[])
then try
screen.DrawImage(drivers.smileyface, 0, 0);

**Reply by tails1154** [reply]
It should. My code is here for the file loading
```
 var smileyfacefile = Gui.VirtualFileSystem.GetFile(@"1:\Resources\smileyface.bmp");
 var bytes = new byte[1024];
 smileyfacefile.GetFileStream().Read(bytes);
 Bitmap smileyface = new Bitmap(bytes);
```

Edit 1: I tried the `screen.DrawImage(drivers.smileyface, 0, 0);` and it seemed to do nothing.
Edit 2: If it has anything to do with it, the image size is 36 x 36

**Comment by Demiomad** [comment, marked as answer]
Try putting your bitmap outside the isoFiles folder and putting it in another folder, setting that file to Embedded Resource, and doing this:
```cs
// Make sure to include this: using IL2CPU.API.Attribs
[ManifestResourceStream(ResourceName = "projectname.folder.filename.extension") public static byte[] smileyface;

// Bitmap initialization code
Bitmap smiley = new Bitmap(smileyface);
```

**Reply by tails1154** [reply]
Oh, Thanks for helping but I've well lost the source code for this lol

**Reply by Demiomad** [reply]
Ah, alright.

**Reply by Demiomad** [reply]
Had to fix it as the "ILCPU.Attribs" was incorrect.

**Reply by Demiomad** [reply]
You could also use File.ReadAllBytes(@"1:\Resources\smileyface.bmp");
